### PR TITLE
Chore: Migrate MessagesView to Typescript

### PR DESCRIPTION
**Pull request by AlexAlexandre**
## Proposed changes
The proposal of this PR is migrate `MessagesView` to TypeScript.

## Issue(s)	
N/A

## How to test or reproduce
On the app, check if the items below works normally:
* messages list;
* file messages list;
* mentions messages list;
* starred messages list;
* pinned messages list;

## Screenshots
N/A

## Types of changes

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] Improvement (non-breaking change which improves a current function)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Documentation update (if none of the other choices apply)

## Checklist

- [x] I have read the [CONTRIBUTING](https://github.com/RocketChat/Rocket.Chat/blob/develop/.github/CONTRIBUTING.md#contributing-to-rocketchat) doc
- [x] I have signed the [CLA](https://cla-assistant.io/RocketChat/Rocket.Chat.ReactNative)
- [x] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works (if applicable)
- [ ] I have added necessary documentation (if applicable)
- [ ] Any dependent changes have been merged and published in downstream modules

## Further comments
N/A

### CODE REVIEW

File: app/views/MessagesView/index.tsx [hunk 1]
```diff
@@ -48,7 +77,7 @@ class MessagesView extends React.Component {
 		this.load();
 	}
 
-	shouldComponentUpdate(nextProps, nextState) {
+	shouldComponentUpdate(nextProps: any, nextState: any) {
```
Comment (gerzonc) [inline app/views/MessagesView/index.tsx:80]: `shouldComponentUpdate` params actually are supposed to have the same properties as the component, you can use `IMessagesViewProps` as `nextProps` type, for example.

File: app/views/MessagesView/index.tsx [hunk 2]
```diff
@@ -73,17 +102,17 @@ class MessagesView extends React.Component {
...
 			return;
 		}
 		navigation.navigate('RoomInfoView', navParam);
 	};
 
-	jumpToMessage = async ({ item }) => {
+	jumpToMessage = async ({ item }: { item: IMessageItem }) => {
 		const { navigation, isMasterDetail } = this.props;
-		let params = {
+		let params: any = {
```
Comment (gerzonc) [inline app/views/MessagesView/index.tsx:115]: Can't we create an `IParams` interface for this? It seems it has values by default so we are able to know which values it receives.

File: app/views/MessagesView/index.tsx [hunk 3]
```diff
@@ -107,9 +136,9 @@ class MessagesView extends React.Component {
 		}
 	};
 
-	defineMessagesViewContent = name => {
+	defineMessagesViewContent = (name?: string) => {
```
Comment (gerzonc) [inline app/views/MessagesView/index.tsx:139]: This is not supposed to be optional.

File: app/views/MessagesView/index.tsx [hunk 4]
```diff
@@ -227,7 +257,7 @@ class MessagesView extends React.Component {
 		}
 	};
 
-	getCustomEmoji = name => {
+	getCustomEmoji = (name: any) => {
```
Comment (gerzonc) [inline app/views/MessagesView/index.tsx:260]: `name: { [name: string]: string }` ?

File: app/views/MessagesView/index.tsx [hunk 5]
```diff
@@ -257,8 +287,8 @@ class MessagesView extends React.Component {
 		try {
 			const result = await this.content.handleActionPress(message);
 			if (result.success) {
-				this.setState(prevState => ({
-					messages: prevState.messages.filter(item => item._id !== message._id),
+				this.setState((prevState: any) => ({
```
Comment (gerzonc) [inline app/views/MessagesView/index.tsx:290]: `IState`

File: app/views/MessagesView/index.tsx [hunk 6]
```diff
@@ -306,12 +336,13 @@ class MessagesView extends React.Component {
...
-const mapStateToProps = state => ({
+const mapStateToProps = (state: any) => ({
 	baseUrl: state.server.server,
 	user: getUserSelector(state),
 	customEmojis: state.customEmojis,
 	useRealName: state.settings.UI_Use_Real_Name,
 	isMasterDetail: state.app.isMasterDetail
 });
 
+// @ts-ignore
```
Comment (gerzonc) [inline app/views/MessagesView/index.tsx:347]: Why?

**Review by gerzonc — CHANGES_REQUESTED** [review]

File: app/views/MessagesView/index.tsx (hunk 4 above)
Comment (AlexAlexandre) [reply app/views/MessagesView/index.tsx:260]: to resolve that, I needed change the `customEmojis` to receive an index signature... 
the code was like this: `customEmojis: { [key: string]: string };` and the `getCustomEmoji = (name: string) => {`

File: app/views/MessagesView/index.tsx (hunk 6 above)
Comment (AlexAlexandre) [reply app/views/MessagesView/index.tsx]: When I started the typescript migration, I didn't have all the knowledge about the code base.
So the `app/containers/ActionSheet/Provider.tsx` file only received `Functional Components` but in fact we use it with class components too. 

I was not planning do this change in this PR, to do it separate in the future, but I fixed. 
Thanks.

**Review by diegolmello — APPROVED** [review]

**Pull request merged**